patch sequence page (#1205)
ocaml/v3.ocaml.org: data/tutorials/gs_00_up_and_running.md

This page will walk you through the installation of everything you need for a comfortable development environment to write projects in OCaml code. Of course, this includes [installing the compiler](#installing-ocaml) itself, but it also installs a build system, a package manager, an LSP server to support your editor, and a few other tools that we describe [later](#setting-up-development-tools), setting up [editor support](#configuring-your-editor), and bootstrapping a [new project](#starting-a-new-project).

We provide installation instructions for Linux, macOS, and *BSD for all OCaml versions. For Windows, we provide instructions on this page for installing OCaml 4.14.0 via the [Diskuv OCaml](https://github.com/diskuv/dkml-installer-ocaml#readme) Installer. Note that, if you use Windows Subsystem for Linux (WSL), the Unix instructions can be used on Windows.

If you are setting up OCaml on Windows and are unsure which installation method to use, you might be
interested in reading [OCaml on Windows](/docs/ocaml-on-windows) first.

> **Guidelines for Following Instructions on this Page**

Reviewer: ```suggestion
**Guidelines for Following Instructions on This Page**
```

## Installing OCaml

> OCaml has an official package manager, `opam`, which allows you to

Reviewer: ```suggestion
OCaml has an official package manager, opam, which allows you to
```

> easily switch between OCaml versions and and much more.

Reviewer: ```suggestion
easily switch between OCaml versions and much more.
```

> It enables you to conveniently deal with different projects which

Reviewer: ```suggestion
It enables you to conveniently deal with different projects that
```

require different versions of OCaml.

> [`opam`](https://opam.ocaml.org/) introduces the concept of a "switch," which is an isolated environment that contains an OCaml compiler together with a set of packages (libraries and other files). Switches allow us to install independent sets of dependencies for different projects.

Reviewer: ```suggestion
[opam](https://opam.ocaml.org/) introduces the concept of a "switch," which is an isolated environment that contains an OCaml compiler together with a set of packages (libraries and other files). Switches allow us to install independent sets of dependencies for different projects.
```

However, please note that it is still experimental and in active development
(any feedback is highly appreciated).

### Installation on Unix (Linux, macOS, *BSD)

Note: OCaml is available as a package in most Linux distributions; however, it is

> often outdated. The best way to install OCaml is with `opam`, OCaml's package manager.

Reviewer: ```suggestion
often outdated. The best way to install OCaml is with opam, OCaml's package manager.
```

The following steps depend on having these packages or tools installed:
`gcc`, `build-essential`, `curl`, `bubblewrap`, and `unzip`.

#### 1. Install opam

> To install opam, you can [use your system package manager](https://opam.ocaml.org/doc/Install.html#Using-your-distribution-39-s-package-system) or download the [binary distribution](https://opam.ocaml.org/doc/Install.html#Binary-distribution). The details are available in the above links, but for convenience, we copy a few of them here:

Reviewer: ```suggestion
To install opam, you can [use your system package manager](https://opam.ocaml.org/doc/Install.html#Using-your-distribution-39-s-package-system) or download the [binary distribution](https://opam.ocaml.org/doc/Install.html#Binary-distribution). The details are available in the above links, but for convenience, we copied a few of them here:
```

Now check the installation by running `opam --version`. You can compare it with the current version on [opam.ocaml.org](https://opam.ocaml.org/).

**Please note:** In case you are running `opam init` inside a Docker container,

> you will be asked whether you want to disable sanboxing.

Reviewer: ```suggestion
you will be asked whether you want to disable sandboxing.
```

This is necessary, unless you run a privileged Docker container.

> #### 3. Create an `opam` Switch

Reviewer: ```suggestion
#### 3. Create an opam Switch
```

This step is necessary only if you need to install a specific version of OCaml,
or if you want to create a new independent environment.

> `opam init` already set up a default `opam` switch for you to work in.

Reviewer: ```suggestion
`opam init` already set up a default opam switch for you to work in.
```

In case you are not satisfied with the OCaml version of your system switch, you can change the version with `opam switch create <version_here>`. More information can be found on the [official website](https://opam.ocaml.org/).

### Installation on Windows

In this section, we'll describe using the [Diskuv OCaml](https://github.com/diskuv/dkml-installer-ocaml#readme) ("DKML") Windows installer. Expect to see another officially-supported Windows installation provided directly by opam in the coming months; it will be compatible with your DKML installation.

Note that only OCaml version 4.14.0 is available via Diskuv OCaml.

> Advanced Users: If you are familiar with Cygwin or WSL2, there are other installation methods described on the [OCaml on Windows](/docs/ocaml-on-windows) page.

#### 1. Use the DKML Installer

Before using the DKML installer, briefly review the following:

> * Do not use the installer if you have a space in your username (ex. `C:\Users\Jane Smith`).

Reviewer: ```suggestion
* Do not use the installer if you have a space in your username (e.g., `C:\Users\Jane Smith`).
```

Now, download and run:

* OCaml 4.14.0 with Git and Visual Studio compiler: [setup-diskuv-ocaml-windows_x86_64-1.2.0.exe](https://github.com/diskuv/dkml-installer-ocaml/releases/download/v1.2.0/setup-diskuv-ocaml-windows_x86_64-1.2.0.exe)

> #### 2. Create an `opam` Switch

Reviewer: ```suggestion
#### 2. Create an opam Switch
```

To learn more about Diskuv OCaml, see the [official
Diskuv OCaml documentation](https://diskuv-ocaml.gitlab.io/distributions/dkml/#introduction).

> ### The OCaml Base Tools are Now Installed

Reviewer: ```suggestion
### The OCaml Base Tools Are Now Installed
```
